projects/manager/src/app/datasets/dataset-editor/dataset-editor.component.ts

```diff
@@ -85,6 +97,60 @@ export class DatasetEditorComponent implements OnChanges {
         }
     }
 
+    removeTag(tag: string): void {
+        const tags: Array<string> = this.form.controls.tags.value;
+
+        const index = tags.indexOf(tag);
+        if (index > -1) {
+            tags.splice(index, 1);
+        }
+
+        this.form.markAsDirty();
+    }
+
+    addTag(event: MatChipInputEvent): void {
+        const tags: Array<string> = this.form.controls.tags.value;
+        const tag = event.value;
+        const input = event.input;
+
+        if (!isValidTag(tag)) {
+            return;
+        }
+
+        input.value = '';
+
+        tags.push(tag);
+
+        this.form.markAsDirty();
+    }
+
+    async saveProvenance(): Promise<void> {
+        if (!this.provenanceComponent.form.valid) {
+            return;
+        }
+
+        const provenance = this.provenanceComponent.getProvenance();
+
+        try {
+            await this.datasetsService.updateProvenance(this.datasetListing.name, provenance);
+            this.snackBar.open('Dataset provenance successfully updated.', 'Close', {duration: 2000});
```

Comment:
Please add the duration to a config.
You will have several durations in the code.

Reply:
2b32712ae206adc91f4676ea7d6eecf0df7fba24